railties/test/rails_info_controller_test.rb

```diff
@@ -36,11 +36,6 @@ def setup
     assert_select "p"
   end
 
-  test "info controller allows requests when all requests are considered local" do
```

Comment:
I think we can revive this test by setting `Rails.application.config.consider_all_requests_local = true` temporarily and assign `@request.env["REMOTE_ADDR"] = "example.org"`.

It seems it was changed unintentionally by https://github.com/rails/rails/commit/05a6fafdd5559ff53897afb8a2aff353e5ac8805 and https://github.com/rails/rails/commit/05a6fafdd5559ff53897afb8a2aff353e5ac8805.

Reply:
Looks like you linked twice to the same PR, what's the other PR so I an have a look? @simi

Reply:
Damn, I can't find the second link again. Maybe it was https://github.com/rails/rails/commit/f209d3898fbd866e1405861319b85c97674a0508 originally introducing one. :( Anyway if I understand it well, this test is covering scenario when reaching related controller outside of local network (like `127.0.0.1`) and has `consider_all_requests_local` enabled. Due to some refactorings in the history both assumptions were most likely removed by mistake.

Reply:
Correct me if I'm wrong. I did some digging and `Rails.application.config.consider_all_requests_local` is unrelated to this. `@request.env["REMOTE_ADDR"]` is more about a request's IP as seen in https://github.com/rack/rack/blob/main/lib/rack/request.rb#L414 while `Rails.application.config.consider_all_requests_local` is related to [nonce generation](https://github.com/rails/rails/pull/32018/files#diff-6fc9b45c9f8ff2375f418deeda05613043b2d654b9fd69177634c3efaeaf66a7)  and possibly what kind of errors are displayed when something fails.

Moreover, it looks like`Rails.application.config.consider_all_requests_local` does nothing here.

<img width="718" alt="Screenshot 2022-06-12 at 20 35 37" src="https://user-images.githubusercontent.com/100293177/173248148-6915f080-bf6f-460a-9dc3-0387ba383386.png">

Reply:
I'm not sure about current implementation, but it is testing this controller -> https://github.com/rails/rails/blob/a4662721965e127bfd7de2192dbd829339ab2058/railties/lib/rails/info_controller.rb#L6 based (and secured) by https://github.com/rails/rails/blob/a4662721965e127bfd7de2192dbd829339ab2058/railties/lib/rails/application_controller.rb#L15-L23

and this one particular test (guessing by the name and history) is testing following scenario:

- `Rails.application.config.consider_all_requests_local #=> true`
- `request.local?  #=> false`

Reply:
OK I think I get what "local" means in both contexts. Does this fix the test? @simi. PR updated.

Reply:
> I'm not sure about current implementation, but it is testing this controller ... based (and secured) by ...

To add some more history, the `Rails.application.config.consider_all_requests_local` check was factored out from `Rails::InfoController` to `Rails::ApplicationController` in 1602a70fb450aba9b5e5befe4c7bfc3344471a06.

That said, I'm not sure if this test is valuable for `Rails::InfoController` specifically, and it might be better factored out into a separate `railties/test/rails_application_controller_test.rb` file (as a test of `before_action :require_local!`).